cmd/createtree/main_test.go

```diff
@@ -43,147 +61,113 @@ func marshalAny(p proto.Message) *any.Any {
 }
 
 func TestRun(t *testing.T) {
-	err := os.Chdir("../..")
-	if err != nil {
-		t.Fatalf("Unable to change working directory to ../..: %s", err)
-	}
-
-	pemPath, pemPassword := "testdata/log-rpc-server.privkey.pem", "towel"
-	pemSigner, err := keys.NewFromPrivatePEMFile(pemPath, pemPassword)
-	if err != nil {
-		t.Fatalf("NewFromPrivatePEM(): %v", err)
-	}
-	pemDer, err := keys.MarshalPrivateKey(pemSigner)
-	if err != nil {
-		t.Fatalf("MarshalPrivateKey(): %v", err)
-	}
-	anyPrivKey, err := ptypes.MarshalAny(&keyspb.PrivateKey{Der: pemDer})
-	if err != nil {
-		t.Fatalf("MarshalAny(%v): %v", pemDer, err)
-	}
-
-	// defaultTree reflects all flag defaults with the addition of a valid pk
-	defaultTree := &trillian.Tree{
-		TreeState:          trillian.TreeState_ACTIVE,
-		TreeType:           trillian.TreeType_LOG,
-		HashStrategy:       trillian.HashStrategy_RFC6962_SHA256,
-		HashAlgorithm:      sigpb.DigitallySigned_SHA256,
-		SignatureAlgorithm: sigpb.DigitallySigned_RSA,
-		PrivateKey:         anyPrivKey,
-		MaxRootDuration:    ptypes.DurationProto(0 * time.Millisecond),
-	}
-
-	server, lis, stopFn, err := startFakeServer()
-	if err != nil {
-		t.Fatalf("Error starting fake server: %v", err)
-	}
-	defer stopFn()
-
-	validOpts := newOptsFromFlags()
-	validOpts.addr = lis.Addr().String()
-	validOpts.pemKeyPath = pemPath
-	validOpts.pemKeyPass = pemPassword
-
 	nonDefaultTree := *defaultTree
 	nonDefaultTree.TreeType = trillian.TreeType_MAP
 	nonDefaultTree.SignatureAlgorithm = sigpb.DigitallySigned_ECDSA
 	nonDefaultTree.DisplayName = "Llamas Map"
 	nonDefaultTree.Description = "For all your digital llama needs!"
 
-	nonDefaultOpts := *validOpts
-	nonDefaultOpts.treeType = nonDefaultTree.TreeType.String()
-	nonDefaultOpts.sigAlgorithm = nonDefaultTree.SignatureAlgorithm.String()
-	nonDefaultOpts.displayName = nonDefaultTree.DisplayName
-	nonDefaultOpts.description = nonDefaultTree.Description
-
-	emptyAddr := *validOpts
-	emptyAddr.addr = ""
-
-	invalidEnumOpts := *validOpts
-	invalidEnumOpts.treeType = "LLAMA!"
-
-	invalidKeyTypeOpts := *validOpts
-	invalidKeyTypeOpts.privateKeyType = "LLAMA!!"
-
-	emptyPEMPath := *validOpts
-	emptyPEMPath.pemKeyPath = ""
-
-	emptyPEMPass := *validOpts
-	emptyPEMPass.pemKeyPass = ""
-
-	pemKeyOpts := *validOpts
-	pemKeyOpts.privateKeyType = "PEMKeyFile"
-	pemKeyTree := *defaultTree
-	pemKeyTree.PrivateKey, err = ptypes.MarshalAny(&keyspb.PEMKeyFile{
-		Path:     pemPath,
-		Password: pemPassword,
+	runTest(t, []*testCase{
```

Comment:
The convention seems to prefer "regular" table-driven, but it may be because `t.Run` is relatively recent addition, iirc. Maybe we should take this question to stylists? Meanwhile, I'd vouch for the simplest approach (no `t.Run` unless you really need it).